tests/security/selinux/selinux_smoke.pm

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Test with SELinux is "enabled" and in "permissive" mode the
+#          system still work.
+# Maintainer: llzhao <llzhao@suse.com>
+# Tags: poo#40361, tc#1682591
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use registration "add_suseconnect_product";
+use version_utils "is_sle";
+
+sub run {
+    select_console "root-console";
+
+    # make sure SELinux is "enabled" and in "permissive" mode
+    validate_script_output("sestatus", sub { m/SELinux\ status: .*enabled.* Current\ mode: .*permissive/sx });
+
+    # refresh & update
+    zypper_call("ref");
+    zypper_call("up");
+
+    # install & remove pkgs, e.g., apache2
+    zypper_call("in apache2");
+    zypper_call("rm apache2");
+
+    # for sle, register available extensions and modules, e.g., free addons
+    if (is_sle) {
+        my $SCC_REGCODE = get_required_var("SCC_REGCODE");
+        assert_script_run("SUSEConnect -r $SCC_REGCODE");
```

Comment:
Is it really necessary? SLE images created in o.s.d are usually already registered.
Also why it is not visible in verification run http://10.67.19.89/tests/349#step/selinux_smoke/10 which was done for sle-12-SP4-Server-DVD?

Reply:
Sorry for the confusing.
1.
I chose using existed image (created by openQA other auto test cases) for my test cases,
so the register is necessary, also, re-register is no harm in my opinion. 

2.
I am revising code according to comments, now the right link is http://10.67.19.89/tests/349#step/selinux_smoke/14